Tests/KituraTests/TestCodableRouter.swift

```diff
@@ -150,8 +150,13 @@ class TestCodableRouter: KituraTest {
             print("POST on /urlencoded for user \(user)")
             respondWith(user, nil)
         }
+        router.post("/optionalurlencoded") { (OptionalUser: OptionalUser, respondWith: (OptionalUser?, RequestError?) -> Void) in
+            print("POST on /urlencoded for user \(OptionalUser)")
+            respondWith(OptionalUser, nil)
+        }
 
         let user = User(id: 4, name: "David")
+        let optionalUser = OptionalUser(id: nil, name: "David")
```

Comment:
What happens if we have this User struct: 
```
struct User: Codable {
  var id: Int?
  var name: String
}
```
and we get `id=4&name=`, we throw an error because `name` cannot be `nil`?

Reply:
Yes. It will treat the empty string as nil even for strings. So if you have a non-optional String variable and you get the empty string it will throw an error.